app/server.R

```diff
@@ -20,301 +13,487 @@ library(plotly) # interactive available data plotting
+  observe({
     
-    # ----------- For the ReadMe HTML -----------
-    # Directly using includeHTML in ui.R will break Shiny (stop execution of everything follows)
+    if (input$main_selector == 'Climate ID'){
+      
+      # Station Climate ID Selection by User
+      updateSelectInput(session, 'stn_id_input',
+                        choices = station_meta()$climate_id[!is.na(station_meta()$climate_id)],
+                        selected = "1168520"
+      )
+      
+    } else if (input$main_selector == 'WMO ID'){
+      
+      # Station WMO ID Selection by User
+      updateSelectInput(session, 'stn_id_input',
+                        choices = station_meta()$WMO_id[!is.na(station_meta()$WMO_id)],
+                        selected = NULL
+      )
+      
+      
+    } else if (input$main_selector == 'TC ID'){
+      
+      # Station TC ID Selection by User
+      updateSelectInput(session, 'stn_id_input',
+                        choices = station_meta()$TC_id[!is.na(station_meta()$TC_id)],
+                        selected = NULL
+      )
+      
+      
+    }
+  }) # End of updating climate ID enter
+  
+  
+  # Translate the 3 IDs to the unified "station_id"
+  id_entered <- reactive({
     
-    output$ReadMe_HTML <- renderUI({
+    validate(
+      need(input$stn_id_input, "Invalid ID Input"))
     
-        includeHTML("Intro_tab.html")
+    if (input$main_selector == 'Climate ID'){
       
-    })
+      station_meta() %>% 
+        filter(climate_id == input$stn_id_input) %>% 
+        select(station_id) %>% slice(1) %>% as.numeric()
+
+    } else if (input$main_selector == 'WMO ID'){
+      
+      station_meta() %>% 
+        filter(WMO_id == input$stn_id_input) %>% 
+        select(station_id) %>% slice(1) %>% as.numeric()
+      
+      
+    } else if (input$main_selector == 'TC ID'){
+      
+      station_meta() %>% 
+        filter(TC_id == input$stn_id_input) %>% 
+        select(station_id) %>% slice(1) %>% as.numeric()
+
+    }
     
+
+  }) # End of translating the 3 IDs to station_id
+  
+  
+  output$stn_input_info <- renderText({
     
+    validate(
+      need(input$stn_id_input, "Invalid Station ID"))
     
-    # ----------- For the Map -----------
+    # Return Station Name & Info
+    map_plot_data() %>% filter(station_id == id_entered()) %>%
+      select(text) %>% as.character()
     
-    # Retrieve all station meta-data
-    current.csv.date <- paste0("station_meta_data.csv") %>% 
-                            file.info() %>%
-                            "$"(ctime) %>% 
-                            base::as.Date()
     
-    output$info_date <- renderText({current.csv.date %>% as.character()})
+  })
+  
+  # Sometimes monthly interval has shorter record length
+  output$stn_warning <- renderText({
     
-    # Update meta-data list if there's no file or the file is older than 7 days
-    if(is.na(current.csv.date) || (Sys.Date() - current.csv.date) > 7) {
```

Comment:
is this not being done? someone is going to click the update button for sure everytime they're on the page...